onmt/translate/translation_server.py

```diff
@@ -540,13 +541,11 @@ def run(self, inputs):
 
         if len(texts_to_translate) > 0:
             try:
-                scores, predictions = self.translator.translate(
-                    texts_to_translate,
-                    src_feats=texts_features,
-                    tgt=texts_ref,
-                    batch_size=len(texts_to_translate)
-                    if self.opt.batch_size == 0
-                    else self.opt.batch_size)
+                infer_iter = textbatch_to_tensor(self.vocabs,
+                                                 texts_to_translate)
```

Comment:
This looks like an oversimplification. I might be missing something but it seems we are loosing the potential configurability of batch_size and the `tgt` stuff (useful for continuing translations with target prefix IIRC).

Reply:
Yes and also the features. see my other comment on textbatch_to_tensor.

Reply:
@anderleich I indeed oversimplified here because I don't understand the way you pass features from the server to the translate() function. It seems odd compared to the structure of the Json batch.
https://github.com/vince62s/OpenNMT-py/blob/removefields/onmt/inputters/text_utils.py#L37-L110